lib/src/layout_grid.dart

```diff
@@ -6,60 +6,82 @@ import 'package:figma_layout_grid/src/rows.dart';
 import 'package:flutter/material.dart';
 
 class LayoutGrid extends StatefulWidget {
-  const LayoutGrid({
+  LayoutGrid({
     super.key,
     required this.builder,
     this.columnsParams = const ColumnsParams(),
     this.rowsParams = const RowsParams(),
     this.gridParams = const GridParams(),
-  });
+    LayoutGridNotifier? notifier,
+  }) {
+    this.notifier = notifier ?? LayoutGridNotifier();
+  }
 
   final ColumnsParams columnsParams;
   final RowsParams rowsParams;
   final GridParams gridParams;
 
   final Widget Function(BuildContext context) builder;
 
+  late final LayoutGridNotifier notifier;
+
   @override
   State<LayoutGrid> createState() => _LayoutGridState();
+
+  static LayoutGridNotifier of(BuildContext context) {
+    final notifier = context
+        .dependOnInheritedWidgetOfExactType<LayoutGridController>()
+        ?.notifier;
+    if (notifier != null) {
+      return notifier;
+    }
+    throw FlutterError(
+        'Build context passed to the of method does not contain the specified widget. Ensure that the widget tree contains a widget of the specified type');
+  }
 }
 
 class _LayoutGridState extends State<LayoutGrid> {
-  late LayoutGridController controller;
-
   @override
   void initState() {
     super.initState();
-    controller = LayoutGridController();
-    controller.addListener(() {
+    widget.notifier.addListener(() {
+      setState(() {});
+    });
+  }
+
+  @override
+  void didUpdateWidget(covariant LayoutGrid oldWidget) {
+    oldWidget.notifier.dispose();
```

Comment:
you shouldn't dispose it here, you should only remove listener that you registered